contactsummary/CRM/Contactsummary/Page/ContactSummary.php

```diff
@@ -16,6 +16,11 @@ private static function registerScripts() {
       ->addVars('contactsummary', array(
         'baseURL' => CRM_Extension_System::singleton()->getMapper()->keyToUrl('org.civicrm.contactsummary')))
       ->addScriptFile('org.civicrm.contactsummary', CRM_Core_Config::singleton()->debug ? 'js/src/contact-summary.js' : 'js/dist/contact-summary.min.js', 1005)
-      ->addStyleFile('org.civicrm.contactsummary', 'css/contactsummary.css');
+      ->addStyleFile('org.civicrm.contactsummary', 'css/contactsummary.css')
+      ->addSettingsFactory(function () {
```

Comment:
the `addSettingsFactory` should only be used if you really need a complex logic/calculation (for example, to build the settings dynamically), otherwise, you can use the `addSetting` method:

```php
->addSetting([
  'FieldOptions' => CRM_Hrjobcontract_Page_JobContractTab::getFieldOptions()
]);
```